src/makeCalculator.js

```diff
@@ -5,6 +5,41 @@
  */
 function makeCalculator() {
   // write code here
+  const calculator = {
+    result: 0,
+    methods: [],
+  };
+
+  calculator.reset = () => {
+    calculator.result = 0;
+
+    return calculator;
+  };
+
+  calculator.add = (number) => (calculator.result += number);
+  calculator.subtract = (number) => (calculator.result -= number);
+  calculator.multiply = (number) => (calculator.result *= number);
+
+  calculator.divide = (number) => {
+    if (number !== 0) {
+      calculator.result /= number;
+    }
+  };
+
+  calculator.methods.push(calculator.add);
+  calculator.methods.push(calculator.subtract);
+  calculator.methods.push(calculator.multiply);
+  calculator.methods.push(calculator.divide);
+
+  calculator.operate = (callback, number) => {
+    if (calculator.methods.includes(callback)) {
+      callback(number);
+    }
```

Comment:
In the `operate` method, you should check if `callback` is a function before calling it. This will prevent runtime errors if a non-function is passed as the `callback`.